src/gvmi/gvmiServer.ts

```diff
@@ -0,0 +1,72 @@
+import { GftpStorageProvider } from "../storage";
+import fs from "fs";
+import jsSha3 from "js-sha3";
+import { nullLogger } from "../utils";
+import { GolemConfigError } from "../error/golem-error";
+
+/**
+ * Helper class to serve a local gvmi file so a provider can
+ * download it directly from you. When you start serving the file,
+ * the server will calculate the hash automatically.
+ */
+export class GvmiServer {
+  private gftp: GftpStorageProvider;
+  public readonly fileHash: string;
+  private fileUrl?: string;
+
+  constructor(private gvmiPath: string) {
+    // no need to check if we're in a node environment here because it's already checked in GftpStorageProvider
+    this.gftp = new GftpStorageProvider(nullLogger());
```

Comment:
maybe this should be an option in the constructor? If there are any errors while transferring the file (this sometimes happened), we will not see the logs...